packages/language-service/test/lib-new/edit-time-parser.spec.ts

```diff
@@ -0,0 +1,323 @@
+import {
+    parseForEditing,
+    ERRORS,
+    AMBIGUITY,
+} from '@stylable/language-service/dist/lib-new/edit-time-parser';
+import {
+    assertRule,
+    assertAtRule,
+    assertDecl,
+    assertInvalid,
+} from '../test-kit/postcss-node-asserts';
+import { expect } from 'chai';
+import { deindent } from '@stylable/core-test-kit';
+
+function safeParse(source: string) {
+    return parseForEditing(deindent(source));
+}
+
+describe('edit-time-parser', () => {
```

Comment:
do we need to handle unclosed strings or cooments?

Reply:
That's interesting.

I'm actually not sure if there is a case for top level strings in CSS. 
postcss takes top level strings as part of the selector:

```css
"string"
.x {} /*selector is: "string"\n.x */
```

Leaving the string open causes postcss to fail, but the safe-parser handles it, taking the string as part of the selector. Not amazing, but it's resilient.

Open comments don't break the safe-parser either, and simply keep taking content into them until they reach another comment closer or the end-of-file.

-----

I added tests for the cases for the parser.

The completions won't work after both cases: 
 - a comment will take any "valid" syntax after it as text. 
 - a string will become part of the definition that comes after it, either part of the selector, which results in an invalid selector node, or part of the "before" content of a declaration.